src/slices/posts.slice.ts

```diff
@@ -0,0 +1,51 @@
+/* eslint-disable no-param-reassign */
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { Post } from '../types/Post';
+import { getUserPosts } from '../api/posts';
+
+type Posts = {
+  loaded: boolean;
+  hasError: boolean;
+  items: Post[];
+};
+
+const initialState: Posts = {
+  loaded: false,
+  hasError: false,
+  items: [],
+};
+
+export const fetchPosts = createAsyncThunk<Post[], number>(
+  'posts/fetchPosts',
+  async (userId: number) => {
+    const posts = await getUserPosts(userId);
+
+    return posts;
+  },
+);
+
+const postsSlice = createSlice({
+  name: 'posts',
+  initialState,
+  reducers: {
+    resetPost: state => {
+      state.items = [];
+    },
+  },
+  extraReducers: builder => {
+    builder.addCase(fetchPosts.pending, state => {
+      state.loaded = true;
```

Comment:
The 'loaded' state is set to true when the fetchPosts action is pending. It might be more appropriate to use a different state property, such as 'loading', to indicate that the data is currently being fetched. This would help avoid confusion with the 'loaded' state, which is set to false when the data is successfully fetched or if there is an error.